tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -713,19 +718,86 @@ describe("updateTechRecord", () => {
       techRecord.vin = "XMGDE02FS0H012345";
       techRecord.primaryVrm = "JY58FPP";
       try {
-        expect(await techRecordsService.updateTechRecord(techRecord, msUserDetails)).toThrowError();
+        expect(await techRecordsService.updateTechRecord(techRecord, msUserDetails, newStatusCode)).toThrowError();
       } catch (errorResponse) {
         expect(errorResponse).toBeInstanceOf(HTTPError);
         expect(errorResponse.statusCode).toEqual(404);
         expect(errorResponse.body).toEqual(HTTPRESPONSE.RESOURCE_NOT_FOUND);
       }
     });
   });
+
+  context("A technical record status is updated to archived via PUT verb", () => {
+    it("should change the records status to archived and set audit details and create no other record", async () => {
+    const newStatusCode = STATUS.ARCHIVED;
+    const techRecord: any = cloneDeep(records[43]);
+    console.log("New test \n", techRecord.techRecord[0].statusCode);
+    const MockDAO = jest.fn().mockImplementation(() => {
+      return {
+        getBySearchTerm: () => {
+          return Promise.resolve({
+            Items: [cloneDeep(records[43])],
+            Count: 1,
+            ScannedCount: 1
+          });
+        }
+      };
+    });
+    const techRecordsService = new TechRecordsService(new MockDAO());
+    const updatedTechRec: any = await techRecordsService.createAndArchiveTechRecord(techRecord, msUserDetails, newStatusCode);
+    expect(STATUS.ARCHIVED).toEqual(updatedTechRec.techRecord[0].statusCode);
+  });
+  });
+
+  context("A technical record status is updated to current from provisional via PUT verb and a current already exists", () => {
+    it("should change the current record status to archived, and update the provisional to current and set audit details for both", async () => {
+    const newStatusCode = STATUS.CURRENT;
+    const techRecord: any = cloneDeep(records[44]);
+    const MockDAO = jest.fn().mockImplementation(() => {
+      return {
+        getBySearchTerm: () => {
+          return Promise.resolve({
+            Items: [cloneDeep(records[44])],
+            Count: 1,
+            ScannedCount: 1
+          });
+        }
+      };
+    });
+    const techRecordsService = new TechRecordsService(new MockDAO());
+    const updatedTechRec: any = await techRecordsService.createAndArchiveTechRecord(techRecord, msUserDetails, newStatusCode);
+    expect(STATUS.CURRENT).toEqual(updatedTechRec.techRecord[2].statusCode);
+  });
+  });
+  context("A technical record status is updated to current from provisional via PUT verb and no current record exists", () => {
+    it("should change the current record status to archived, and update the provisional to current and set audit details for both", async () => {
```

Comment:
in context block it says "and no current record exists" and then in the IT block "should change the current to archived". Description of the test case should change to match accordingly.

Also, I see you;'re checking only if the vehicle has a current, you should also check that the provisional is now archived. The same above, check that the current and provisional are now archived as well